lambdas/service/app.py

```diff
@@ -831,36 +836,79 @@ def get_order():
-          type: string
-          description: The desired format of the output. Possible values are `compact` (the default) for a tab-separated
-          manifest and `terra.bdbag` for a manifest in the format documented `http://bd2k.ini.usc.edu/tools/bdbag/. The
-          latter is essentially a ZIP file containing two manifests: one for participants (aka Donors) and one for
-          samples (aka specimens). The format of the manifests inside the BDBag is documented here:
-          https://software.broadinstitute.org/firecloud/documentation/article?id=10954
-        - name: token
-          in: query
-          type: string
-          description: Reserved. Do not pass explicitly.
+                - `compact` (the default) for a compact tab-separated manifest
 
-    :return: If the manifest generation has been started or is still ongoing, the response will have a
-    301 status and will redirect to a URL that will get a recheck the status of the manifest.
+                - `full` for a full tab-separated manifest
 
-    If the manifest generation is done and the manifest is ready to be downloaded, the response will
-    have a 302 status and will redirect to the URL of the manifest.
-    """
+                - `terra.bdbag` for a manifest in the format documented
+                  [http://bd2k.ini.usc.edu/tools/bdbag/]. This provides a ZIP file
+                  containing two manifests: one for participants (aka Donors) and one
+                  for samples (aka specimens). The format of the manifests inside the
+                  BDBag is documented here:
+                  [https://software.broadinstitute.org/firecloud/documentation/article?id=10954]
+            ''',
+        ),
+        params.query(
+            'token',
+            schema.optional({'type': 'string'}),
+            description='Reserved. Do not pass explicitly.',
+        ),
+    ],
+}
+
+manifest_response_headers_spec = {
+    'headers': {
+        'Retry-After': {
+            'description': 'Number of seconds to wait before the next request.',
+            'schema': {'type': 'string'},
+        },
+        'Location': {
+            'description': 'URL to be requested next.',
+            'schema': {'type': 'string', 'format': 'url'},
+        }
+    }
+}
+
+
+@app.route('/manifest/files', methods=['GET'], cors=True, path_spec=copy.copy(manifest_path_spec), method_spec={
+    'tags': ['Manifests'],
+    'summary': 'Describe current version of the Azul service',
+    'description': format_description('''
+        Initiate and check status of a manifest generation job, returning
+        either a 301 response redirecting to a URL to re-check the status of
+        the manifest generation or a 302 response redirecting to the location
+        of the completed manifest.
+    '''),
+    'responses': {
+        '301': {
+            'description': 'The manifest generation has been started or is still ongoing',
+            **manifest_response_headers_spec
```

Comment:
I think we should denormalize this i.e., undo the extraction of the variable. The URL in the Location header has a different meaning and its confusing to describe both cases identically (and therefore unnecessarily generic) as URL to be requested next.